.github/workflows/ci-snapshot-publish.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright 2022 JanusGraph Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Publish packages to GitHub Packages
+on:
+  push:
+    branches:
+      - master
+      - v*
+
+jobs:
+  snapshot-publish:
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 8
+          distribution: zulu
+
+      - name: Setup unique version
+        run:  mvn versions:set -DnewVersion="$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)-$(git rev-parse --short HEAD)" -DgenerateBackupPoms=false
+
+      - name: Compile and prepare snapshot release
+        run: mvn clean install -Pjanusgraph-release -Dgpg.skip=true -DskipTests=true
```

Comment:
Wouldn't it be better to not skip GPG signing here (and in the deploy step below)? Or is there any reason why we cannot do that here?